auth-agents-common/src/main/java/org/apache/atlas/policytransformer/PersonaCachePolicyTransformer.java

```diff
@@ -104,19 +105,14 @@ public List<AtlasEntityHeader> transform(AtlasEntityHeader atlasPolicy) {
                 header.setAttribute(ATTR_POLICY_CONDITIONS, templatePolicy.getPolicyConditions());
 
                 if (POLICY_SERVICE_NAME_ABAC.equals(policyServiceName)) {
-                    if (policyFilterCriteria != null && !policyFilterCriteria.isEmpty()) {
-                        try {
-                            JsonNode filterCriteriaNode = mapper.readTree(policyFilterCriteria);
-                            if (filterCriteriaNode != null && filterCriteriaNode.get("entity") != null) {
-                                JsonNode entityFilterCriteriaNode = filterCriteriaNode.get("entity");
-                                policyFilterCriteria = entityFilterCriteriaNode.toString();
-                            }
-                        } catch (JsonProcessingException e) {
-                            LOG.error("ABAC_AUTH: PersonaCachePolicyTransformer.transform: parsing filterCriteria failed for policy={}, filterCriteria={}", atlasPolicy.getGuid(), policyFilterCriteria);
-                        }
+                    if (JsonToElasticsearchQuery.parseFilterJSON(policyFilterCriteria, "entity") == null) {
+                        LOG.error("PolicyRefresher: PersonaCachePolicyTransformer: error parsing policyFilterCriteria for policyId={}, policyFilterCriteria={}", header.getGuid(), policyFilterCriteria);
+                        continue;
                     }
-                    header.setAttribute(ATTR_POLICY_FILTER_CRITERIA,
-                            templatePolicy.getPolicyFilterCriteria().replace(PLACEHOLDER_FILTER_CRITERIA, policyFilterCriteria));
+                    header.setAttribute(ATTR_POLICY_FILTER_CRITERIA, templatePolicy.getPolicyFilterCriteria() != null
```

Comment:
IMO, having policyFilterCriteria in templatePolicy makes sense, let us remove `policyFilterCriteria` field from PolicyTransformerTemplate itself, it will not save from code complexity which we will never use.

It should be always atlasPolicy.filterCriteria as replacements in template filterCriteria will further makes it complex for no reason.
We can add it back if ever needed